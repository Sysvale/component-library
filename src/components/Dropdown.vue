<template>
	<multiselect
		v-bind="$attrs"
		v-model="selectedValue"
		select-label=''
		deselect-label=''
		selected-label=''
		:close-on-select="false"
		:multiple="true"
		:taggable="true"
		:clear-on-select="false"
		:searchable="false"
		@select="selectItem"
		@remove="selectItem"
	>
		<template
			slot="option"
			slot-scope="props"
		>
			<div
				class="option__desc"
			>
				<span class="option__title">
					<span class="d-flex align-self-center align-items-center align-content-center">
						<b-form-checkbox
							id="checkbox-1"
							v-model="props.option.is_selected"
							name="checkbox-1"
							:value="true"
							:unchecked-value="false"
						/>

						{{ props.option.title }}

					</span>
				</span>
			</div>
		</template>
	</multiselect>
</template>

<script>
export default {
	data() {
		return {
			selectedValue: this.$attrs.value,
		}
	},

	methods: {
		selectItem (tag) {
			tag.is_selected = !tag.is_selected;
		},
  }
};
</script>
<style src="vue-multiselect/dist/vue-multiselect.min.css" />
<style>
.multiselect__option--highlight {
	background: #F5F6F8;
	outline: none;
	color: #142032;
}

.custom-control-input:checked ~ .custom-control-label::before {
	color: #fff;
	border-color: #285AB5;
	background-color: #285AB5;
}

.multiselect__tag {
	background: #e9eef1;
	color: #142032;
	border: 1px solid #ced4da;
}

	
.multiselect__tag-icon:after{
	color: #4F606F;
}

.multiselect__tag-icon:focus,
.multiselect__tag-icon:hover {
	background: #d1d9e9;
	color: #4F606F;
}

.multiselect__tag-icon:focus:after,
.multiselect__tag-icon:hover:after {
	color: #4F606F;
}

.multiselect__option--selected.multiselect__option--highlight {
	background: #fff;
	color: #142032;
}
.multiselect__option--selected.multiselect__option--highlight:after {
	background: #fff;
	content: attr(data-deselect);
	color: #142032;
}

.multiselect__option--selected {
	background: #FFF;
	color: #142032;
	font-weight: 600;
}
.multiselect--disabled {
  background: transparent;
}

.multiselect--disabled .multiselect__tags {
	background: #e9eef1 !important;
}
</style>
