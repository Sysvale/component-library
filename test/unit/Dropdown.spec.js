import { shallowMount, mount, createLocalVue, } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue';
import Dropdown from '../../src/components/Dropdown.vue';
import Multiselect from 'vue-multiselect';

const localVue = createLocalVue();
localVue.use(BootstrapVue);
localVue.component('multiselect', Multiselect);

const mockedData = [
	{ title: 'Avengers', is_selected: false, },
	{ title: 'Naruto', is_selected: false, },
	{ title: 'X-men', is_selected: false, },
	{ title: 'The Powerpuff Girls', is_selected: false, }
];

test('Component is mounted properly', () => {
	const wrapper = shallowMount(Dropdown, {
		localVue,
		propsData: {
			options: mockedData,
		},
	});
	expect(wrapper).toMatchSnapshot();
});

describe("Custom checkbox tests", () => {
	test('If the number of the custom checkboxes match the number of elements in the option array', () => {
		const wrapper = mount(Dropdown, {
			localVue,
			propsData: {
				options: mockedData,
				value:[],
				label: 'title',
				trackBy: 'title',
			},
		});

		expect(wrapper.findAll('input').length)
			.toBe(mockedData.length);

		expect(wrapper.findAll('label').length)
			.toBe(mockedData.length);
	});

	test('If the input event is emited properly when the data passed as a v-model changes', () => {
		const wrapper = mount(Dropdown, {
			localVue,
			propsData: {
				options: mockedData,
				value:[],
				label: 'title',
				trackBy: 'title',
			},
		});

		wrapper
			.find(`#checkbox-${mockedData[0].title}`)
			.trigger('click');

		wrapper
			.vm
			.$options
			.watch
			.selectedValue
			.call(wrapper.vm, mockedData[0]);
			
		expect(wrapper.emitted().input)
			.toBeTruthy();

		expect(wrapper.emitted().input)
			.toEqual([[mockedData[0]]]);
	});
});