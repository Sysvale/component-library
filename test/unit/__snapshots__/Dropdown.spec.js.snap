// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component is mounted properly 1`] = `<span id="dropdown" options="[object Object],[object Object],[object Object],[object Object]"><multiselect-stub internalsearch="true" options="[object Object],[object Object],[object Object],[object Object]" multiple="true" value="" placeholder="Select option" allowempty="true" customlabel="function (t,e){return i(t)?&quot;&quot;:e?t[e]:t}" taggable="true" tagplaceholder="Press enter to create a tag" tagposition="top" optionslimit="1000" blockkeys="" showpointer="true" optionheight="40" name="" selectlabel="" selectgrouplabel="Press enter to select group" selectedlabel="" deselectlabel="" deselectgrouplabel="Press enter to deselect group" showlabels="true" limit="99999" maxheight="300" limittext="function (t){return&quot;and &quot;.concat(t,&quot; more&quot;)}" opendirection="" shownooptions="true" shownoresults="true" tabindex="0"></multiselect-stub></span>`;
